pkgs/development/interpreters/lambda-lisp/sources.nix

```diff
@@ -0,0 +1,47 @@
+let
+  lambdaLispVersion = "2022-08-18";
+  blcVersion = "2";
+  justineLolMirror = "https://justine.lol/lambda/";
+in
+{ fetchFromGitHub, fetchurl }: {
+  inherit blcVersion;
+  inherit lambdaLispVersion;
+
+  src = fetchFromGitHub {
+    owner = "woodrush";
+    repo = "lambdalisp";
+    rev = "2119cffed1ab2005f08ab3cfca92028270f08725";
+    sha256 = "sha256-ml2xQ8s8sux+6GwTw8mID3PEOcH6hn8tyc/UI5tFaO0=";
+  };
+
+  uniCSrc = fetchurl {
+    url = "https://tromp.github.io/cl/uni.c";
```

Comment:
could change at any time :/

Reply:
Do you know anywhere I could host it permanently or smth? I'm not sure what to do about situations like this for nixpkgs...

Reply:
It's actually very tiny:

```c
#include <stdio.h>
#include <stdlib.h>
#include <assert.h>
enum {I,O,V,A,L};
int n=44,i,c,T[M]={L,A,8,A,2,  V,0,L,L,V,
    A,30,L,A,2,V,0,L,A,5,A,7,L,V,0,O,
    A,14,L,A,2,V,0,L,A,5,A,2,  V,0,O,O,A},b,s;
typedef struct _{int t,r; struct _*e,*n;} C;C*e,*f,*l,*S[M];
void x(int l,int u){for(;l<=u;T[n++]=T[l++]);}
int g(){i--||(i=b,c=getchar());return c>>i&1;}
void d(C*l){!l||--l->r||(d(l->e),d(l->n),l->n=f,f=l);}
int p(int m){if(g()){for(T[n++]=V;g();T[n]++);n++;}else
          T[m]=n++&&g()?(T[m+1]=p(++n),A):L,p(n);return n-m;}
int main(int t){char o;
 b=t>1?0:7;T[43]=p(n);i=0;
 for(t=b?10:26;;)switch(T[t]){
  case I: g();i++;assert(n<M-99);if(~c&&b){x(0,6);for(T[n-5]=96;i;T[n++]=!g())
           x(0,9);}x(c<0?7:b,9);T[n++]=!b&&!g();break;
  case O: t=b+t>42?(o=2*o|t&1,28):(putchar
              (b?o:t+8),fflush(stdout),b?12:28);break;
  case V: l=e;for(t=T[t+1];t--;e=e->n);
                   t=e->t;(e=e->e)&&e->r++;d(l);break;
  case A: t+=2;f||(f=calloc(1,sizeof(C)));assert(f&&s<M);S[s++]=l=f;f=l->n;
          l->r=1;l->t=t+T[t-1];(l->e=e)&&e->r++;break;
  case L: if(!s--)return 0;S[s]->n=e;e=S[s];t++;break;
 }
 return T[t+2];
}
```

Maybe I could add it to the dir?

Reply:
https://github.com/tromp/tromp.github.io/blob/master/cl/uni.c

Reply:
Okay added that instead, what about 
```
  blcSrc = fetchurl {
    url = "${justineLolMirror}Blc.S?v=${blcVersion}";
    hash = "sha256-qt7vDtn9WvDoBaLESCyyscA0u74914e8ZKhLiUAN52A=";
  };

  flatSrc = fetchurl {
    url = "${justineLolMirror}flat.lds";
    hash = "sha256-HxX+10rV86zPv+UtF+n72obtz3DosWLMIab+uskxIjA=";
  };
```

Are these trustworthy enough?

Reply:
The issue is that the creator could change their contents or the website could disappear

Put them in to archive.org wayback machine

Reply:
Oki I see, didn't know we could use archive.org, nice!